src/App/Tracker/Controller/Hooks/Listeners/AbstractListener.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Part of the Joomla Tracker's Tracker Application
+ *
+ * @copyright  Copyright (C) 2012 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License Version 2 or Later
+ */
+
+namespace App\Tracker\Controller\Hooks\Listeners;
+
+use App\Tracker\Table\IssuesTable;
+use Joomla\Event\Event;
+use Joomla\Github\Github;
+
+use Monolog\Logger;
+
+/**
+ * Abstract listener class for custom Listeners
+ *
+ * @since  1.0
+ */
+abstract class AbstractListener
+{
+	/**
+	 * Check if label already exists
+	 *
+	 * @param   object  $hookData    Hook data payload
+	 * @param   Github  $github      Github object
+	 * @param   Logger  $logger      Logger object
+	 * @param   object  $project     Object containing project data
+	 * @param   string  $checkLabel  The label to check
+	 *
+	 * @return  bool    True if the label already exists
+	 *
+	 * @since   1.0
+	 */
+	protected function checkLabel($hookData, Github $github, Logger $logger, $project, $checkLabel)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueID($hookData);
```

Comment:
If `getIssueId()` will return null as I proposed then we can do a check here like:
```php
if (null === $issueNumber)
{
	$logger->error(
		sprintf(
			'Error retrieving issue number for %s/%s',
			$project->gh_user,
			$project->gh_project
		)
	);

	throw new RuntimeException;
}
```